### DISCUSSION TITLE
Get warnings for both `Ruff` and `Pyright`

### DISCUSSION THREAD

**Discussion by mahdikooshkbaghi** [LazyVim/LazyVim#2565] [Q&A] [answered]
Hi all,
On freshly installed `LazyVim` I have enabled the python language from LazyExtras.
When I open a python script, I am getting both `Pyright` and `Ruff` warnings.
Is is possible ditch `Ruff` in favor of `Pyright`.
Thanks
<img width="1040" alt="Screenshot 2024-02-16 at 11 48 42 PM" src="https://github.com/LazyVim/LazyVim/assets/21984632/d1c199a1-7381-4ca7-b9f6-501e3027ef68">

**Comment by dpetka2001** [comment, marked as answer]
Try this in a file `~/.config/nvim/lua/plugins/lsp.lua`
```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        ruff_lsp = {
          handlers = {
            ["textDocument/publishDiagnostics"] = function() end,
          },
        },
      },
    },
  },
}
```

**Reply by shinonomeow** [reply]
thanks for your answer, its useful for me

**Comment by jmrussell** [comment]
I have the same issue but the above `lsp.lua` doesn't resolve it for me. Any ideas?

**Reply by dpetka2001** [reply]
If you don't want diagnostics from ruff, just disable it as a whole.

**Comment by milkpuff** [comment]
lazyvim now use ruff instead of ruff-lsp, try this. Works for me.
```lua
return {
  "neovim/nvim-lspconfig",
  opts = {
    servers = {
      ruff = {
        init_options = {
          settings = {
            lint = {
              enable = false, 
            },
          },
        },
      },
    },
  },
}

```
